### PR TITLE
issue_69: added 'nth' param to 4 methods; and added tests

### PR DESCRIPTION
**Pull request by weirdtangent**
#cpan-prc #team-ziprecruiter

### CODE REVIEW

**Comment by petdance** [comment]
I think it would be a big help if the docs for the function explained when/why you would use the `nth` param.

**Comment by oalders** [comment]
LGTM

File: lib/WWW/Mechanize.pm [hunk 1]
```diff
@@ -1432,11 +1439,16 @@ sub all_forms_with_fields {
 }
 
 
-=head2 $mech->form_with_fields( @fields )
+=head2 $mech->form_with_fields( @fields, [$nth] )
```
Comment (skaji) [inline lib/WWW/Mechanize.pm]: I'm not very familiar with HTML, but can we assume that field names do not consist of digits only?
That is, can we assume that the following form is invalid?
```html
<form action="http://localhost/" method="post" enctype="multipart/form-data" name="4th_form_2">
    <input type="text" name="4a"  />
    <input type="text" name="2"  /> <!-- DIGIT ONLY FIELD NAME -->
    <input type="Submit" name="submit" value="Submit" label="Sumbit" />
</form>
```

Comment (oalders) [reply lib/WWW/Mechanize.pm]: @jculverhouse did you have thoughts on the comments above?

Comment (oalders) [reply lib/WWW/Mechanize.pm]: @skaji I just checked and https://validator.w3.org doesn't have an issue with the input names in your example form.

**Comment by oalders** [comment]
@petdance any thoughts on these changes?

**Comment by petdance** [comment]
Looks fine to me.

File: lib/WWW/Mechanize.pm (hunk 1 above)
Comment (skaji) [reply lib/WWW/Mechanize.pm]: If `name="\d+"` is valid,
https://gist.github.com/skaji/e6445cb1c33dc89d5be7565d51eb2121 shows that form_with_fields() of this PR is ambiguous.

If we really want to specify "nth" form, I would suggest using a hash `{ n => "nth" }` like select() method.

Comment (oalders) [reply lib/WWW/Mechanize.pm]: Good catch @skaji and also a good suggestion for the API.

**Comment by simbabque** [comment]
This attempts to implement #69. I've rebased and will implement the changes suggested in the discussion on the review. I will submit a new PR based on this one.

**Comment by simbabque** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id says the following:

> Note: Technically, in [HTML5](https://developer.mozilla.org/en-US/docs/Glossary/HTML5), the value for an id attribute may contain any character, except [whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace) characters. However, to avoid inadvertent errors, only [ASCII](https://developer.mozilla.org/en-US/docs/Glossary/ASCII) letters, digits, '_', and '-' should be used and the value for an id attribute should start with a letter. For example, . has a special meaning in CSS (it acts as a class selector). Unless you are careful to escape it in the CSS, it won't be recognized as part of the value of an id attribute. It is easy to forget to do this, resulting in bugs in your code that could be hard to detect.

According to this, in HTML 4 the ID shouldn't have started with a number, but in HTML 5 it's allowed. I agree with @skaji's suggestion to use `{ n => "nth" }` instead. It should work for all of the suggested methods.

**Pull request closed without merging**